.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -359,17 +359,29 @@ private DateTimeResolutionResult ParseComplexDatePeriod(string text, DateObject
                 var isSpecificDate = false;
                 var isStartByWeek = false;
                 var isEndByWeek = false;
+                bool isAmbiguousStart = false, isAmbiguousEnd = false;
+                var ambiguousRes = new DateTimeResolutionResult();
                 var dateContext = GetYearContext(this.config, match.Groups["start"].Value.Trim(), match.Groups["end"].Value.Trim(), text);
 
                 var startResolution = ParseSingleTimePoint(match.Groups["start"].Value.Trim(), referenceDate, dateContext);
 
                 if (startResolution.Success)
                 {
-                    futureBegin = (DateObject)startResolution.FutureValue;
-                    pastBegin = (DateObject)startResolution.PastValue;
-                    isSpecificDate = true;
+                    // Check if the extraction is ambiguous (e.g. "mar" can be resolved to both "March" and "Tuesday" in FR, IT and ES)
+                    if (this.config.AmbiguousPointRangeRegex != null && this.config.AmbiguousPointRangeRegex.IsMatch(match.Groups["start"].Value.Trim()))
```

Comment:
What happens when the term appears by itself now?

Reply:
it is parsed as weekday.

Reply:
Hmmm. Can you add a spec for this case, and add a "Comment" in it explaining the current behaviour? Then we can merge as-is for now.

Reply:
added.